azurelinuxagent/common/event.py

```diff
@@ -393,52 +529,106 @@ def _clean_up_message(message):
-                              DefaultParameter('EventPid', os.getpid() if set_values_for_agent else 0),
-                              DefaultParameter("TaskName", threading.current_thread().getName() if set_values_for_agent else ""),
-                              DefaultParameter("KeywordName", '')]
+        params_to_keep = dict().fromkeys(['Name', 'Version', 'Operation', 'OperationSuccess', 'Message', 'Duration'])
+        trimmed_params = []
+
+        for param in event.parameters:
+            if param.name in params_to_keep:
+                trimmed_params.append(param)
 
-        # Converting the event_parameters into a dictionary as it helps to easily look up and get values
-        param_names = OrderedDict([(param.name, param.value) for param in event_parameters])
+        event.parameters = trimmed_params
+
+    def collect_events(self):
+        """
+        Retuns a list of events that need to sent to the telemetry pipeline and deletes the corresponding files
+        from the events directory.
+        """
+        event_list = TelemetryEventList()
+        event_directory_full_path = os.path.join(conf.get_lib_dir(), EVENTS_DIRECTORY)
+        event_files = os.listdir(event_directory_full_path)
 
-        for param in default_parameters:
-            if param.name not in param_names or set_values_for_agent:
-                # If set_values_for_agent, we disregard any values already set for an existing default property and
-                # replaces it with a latest entry.
-                param_names[param.name] = param.value
+        for event_file in event_files:
+            try:
+                match = EVENT_FILE_REGEX.search(event_file)
+                if match is None:
+                    continue
 
-        parameters = DataContractList(TelemetryEventParam)
-        for name, value in param_names.items():
-            parameters.append(TelemetryEventParam(name, value))
+                event_file_path = os.path.join(event_directory_full_path, event_file)
 
-        return parameters
+                try:
+                    logger.verbose("Processing event file: {0}", event_file_path)
+
+                    with open(event_file_path, "rb") as fd:
+                        event_data = fd.read().decode("utf-8")
+
+                    event = parse_event(event_data)
+
+                    is_legacy_event = match.group('agent_event') is None
+
+                    if is_legacy_event:
+                        # We'll use the file creation time for the event's timestamp
+                        event_file_creation_time_epoch = os.path.getmtime(event_file_path)
+                        event_file_creation_time = datetime.fromtimestamp(event_file_creation_time_epoch)
+
+                        if event.is_extension_event():
+                            EventLogger._trim_extension_event_parameters(event)
+                            self._add_common_event_parameters(event, event_file_creation_time)
```

Comment:
We would want to modify extension events always, right? It doesn't matter if it's a legacy event or not.

Reply:
this is related to the comment I posted above, on line 595 - I need to clarify what I mean by "legacy". This is the comment I added: 

>                   # "legacy" events are events produced by previous versions of the agent (<= 2.2.46) and extensions;
>                   # they do not include all the telemetry fields, so we add them here

Reply:
But my point is that all extension events need to have common parameters added, not just legacy ones? Why is ` self._add_common_event_parameters(event, event_file_creation_time)` only called for legacy extension events?

Reply:
all extension events are legacy events according to my definition.
> is_legacy_event = match.group('agent_event') is None
only (new) agent event events are not legacy.

Reply:
Ah, got it now. Thanks.